Restructure security service page and separate DNS configuration into how-to guide (#194)
emnify/product-docs: docs/services/platform/security.md

## Why does IoT require SASE?

Secure Access Service Edge ([SASE](/glossary#sase)) introduces a new architecture where networking and security functions are bundled in a cloud-delivered service.
This approach allows you to apply consistent security standards to all your devices independent of the location.
Additionally, it enables the integration of security features right from the start.

> Key features of [SASE](/glossary#sase) for IoT architecture includes are:

Reviewer: This was in the original content, but since we're updating anyway let's remove the duplicate glossary link ✏️ 

```suggestion
Key features of SASE for IoT architecture include:
```

- **Dynamic Data Routing with Software-Defined Wide Area Network (SD-WAN):**  
  emnify utilizes a SD-WAN to route data to the closest cloud region using the [Regional Breakout](iot-cloud-communication-platform#regional-breakout) concept.
This enhances latency and data stability, ensuring that data remains within the designated continent and jurisdiction.
- **Cloud Access Security Broker (CASB):**

>   emnify offers centralized policy definition for devices, including networks that can be accessed and authorized IP addresses for remote device access. All configurations are managed through the central platform and applied wherever the device is.

Reviewer: If the formatting allows, it'd be great to keep each sentence on its own line. Also it seems like each point is rendering a different way, so I'd stick with one and keep it consistent.

<details><summary>Screenshot</summary>
<img width="937" alt="Screenshot 2023-10-02 at 20 39 54" src="https://github.com/emnify/product-docs/assets/26869552/0f592565-e312-4709-bd71-649af12f72b7">
</details>

![IoT security threats](assets/security-threats.png)

## Security features

> The following sections will delve into some of the security features offered by emnify.

Reviewer: ```suggestion
The following sections delve into some of the security features offered by emnify.
```

### DNS

When a device establishes a connection, it uses a Domain Name Service (DNS) server to resolve a hostname to an IP address for data transmission.
For example, a hostname like `iot.example.com` is mapped to an IP address like `192.0.2.1`.

> Cellular providers typically provide DNS services, and by default, emnify routes all DNS queries over Google’s public DNS `8.8.8.8`.

Reviewer: I'd revert this change, as we want to keep our sentences as short as possible.

Some devices and modules allow you to configure the DNS service.
For example, Quectel uses the `AT+QIDNSCFG` command, while SIMCom uses the `AT+CDNSCFG` command.

> Configuring the DNS service in this way is beneficial for using your own or private DNS servers, enhancing security and control.

Reviewer: ```suggestion
Configuring the DNS service is beneficial for using your own or private DNS servers, enhancing security and control.
```

See this [guide for how to configure the DNS settings of a public or private DNS server](/services/security/how-to-configure-dns-settings).

<!-- TODO: Recreate dns_setting.png (Custom DNS setting configuration) -->

### IMEI lock

Device manufacturers often face issues related to SIM card theft, as pluggable SIM cards can be removed from a device and then used to gain free internet access.

> The [IMEI lock](/glossary#imei-lock)  feature prevents the use of SIM card in any other device by binding the SIM to a specific [IMEI](/glossary#imei) (International Mobile Equipment Identity), which serves as a unique device identifier.

Reviewer: ```suggestion
The [IMEI lock](/glossary#imei-lock) feature prevents the use of SIM card in any other device by binding the SIM to a specific [international mobile equipment identity (IMEI)](/glossary#imei), which serves as a unique device identifier.
```

> When the automatic IMEI lock is configured, the emnify platform associates the SIM cards to the first device that establishes a data connection.

Reviewer: Lets keep the original wording here:

 ```suggestion
When the automatic IMEI lock is configured, the emnify platform binds the SIM cards to the first device that establishes a data connection.
```

All future device connections are restricted to this specific device.

### Two-factor authentication

> All emnify accounts have the option to enable two-factor authentication (2FA) via the [Portal](https://portal.emnify.com/) to enhance security.

Reviewer: I'd prefer to keep the original wording for this sentence or at least work in the wording of 2FA being an additional layer (as this is descriptive, especially for people who aren't familiar with MFA).
